External/FEXCore/include/FEXCore/Core/CoreState.h

```diff
@@ -8,8 +8,70 @@
 #include <cstddef>
 #include <stdint.h>
 #include <string_view>
+#include <type_traits>
 
 namespace FEXCore::Core {
+  // This is defined here instead of a helper because it is a massive footgun in behaviour.
+  // Don't use this instead of std::atomic unless you really know what you're doing.
+  // Increment and decrement can visibily tear and only uses relaxed atomic behaviour internally.
+  // In particular decrement and increment can tear if a signal is received half-way through
+  // and you need to be aware of that when using this.
```

Comment:
Bit of waffling around here. How about this?

```suggestion
  // Wrapper around std::atomic using std::memory_order_relaxed.
  // This allows compilers to emit more performant code at the expense of visibly tearing.
  // In particular, increments/decrements may visibly tear if a signal is received half-way through.
  //
  // Prefer std::atomic with default memory ordering unless you really know what you're doing.
```

Reply:
Resolved.